### PR TITLE
simulateVCF

### PR DESCRIPTION
**Pull request by rimangan**
Cmd for simulating VCF data. Currently supports gVCF simulation in which allele frequencies are drawn from a stationarity distribution with an input selection parameter. Currently no functionality for position and mutation simulations, but these can be added from the simulateBed and simulate functions in the future.

### CODE REVIEW

File: popgen/AFSsimulate.go
Comment (edotau) [inline popgen/AFSsimulate.go:62]: I think if you return currRow (`func SimulateGenotype(alpha float64, n int, k int) []vcf.GenomeSample {}`) in the function and move the for loop `r := 0; r < k; r++ {}` down to where you write the Vcf, you would not need to allocate memory for the 2-D slice: 
(`var answer [][]vcf.GenomeSample = make([][]vcf.GenomeSample, 0, k)`)
as a result you can write the in the `r := 0; r < k; r++ {}` loop straight to file rather than creating a new currRow every itertation.

**Review by edotau — APPROVED** [review]

File: popgen/AFSsimulate.go
Comment (craiglowe) [inline popgen/AFSsimulate.go:27]: Is "count" counting the number of times that a derived alleles is observed?

File: popgen/AFSsimulate.go
Comment (craiglowe) [inline popgen/AFSsimulate.go:21]: When we say "n individuals" are individuals haploid?

File: popgen/AFSsimulate.go
Comment (craiglowe) [inline popgen/AFSsimulate.go:40]: Maybe throw an error if "n" isn't even?  Seems like we are transitioning here to assuming diploid individuals?

File: popgen/AFSsimulate.go
Comment (craiglowe) [inline popgen/AFSsimulate.go:39]: I don't think this loop helps you because you are not doing: answer[i] = append(answer[i], something)
Later on you run: answer = append(answer[i], something)
I think you can comment out this loop because it is allocating memory that is then not being used.

File: popgen/AFSsimulate.go
Comment (craiglowe) [inline popgen/AFSsimulate.go:45]: I think it makes sense to have the variable declared here, but I don't think you want to allocate the memory (make() after the equals sign) because you are just going to allocate different memory once you enter the loop in a few more lines.

File: popgen/AFSsimulate.go
Comment (craiglowe) [inline popgen/AFSsimulate.go:72]: This comment might not match the current function since it mentions "true" and "false"

**Review by craiglowe — COMMENTED** [review]
Looks good!  I didn't catch anything I thought was a bug, but I had a couple comments.

File: simulate/vcf.go
Comment (craiglowe) [inline simulate/vcf.go:22]: Can we increment the position each time through the counter so that not all SNPs are placed at the same position?

File: simulate/vcf.go
Comment (rimangan) [inline simulate/vcf.go:16]: move this into the loop, make SimulateSite (then check for segregating status)

File: simulate/vcf.go
Comment (rimangan) [reply simulate/vcf.go:22]: pos=i

**Review by cfauci — APPROVED** [review]

**Comment by edotau** [comment]
looks good, but I can't approve this until the merge conflicts are resolved. Can you please make sure to merge the new updates from main?
```
$git checkout main
$git pull
$git checkout rileyStationaritySampler
$git merge
```
then manually resolve the merging conflicts to `numbers/Monte_Carlo.go`, `popgen/AFSsimulate.go` and any other merge conflicts that may arise whenever new code is merged from code review.

**Review by craiglowe — APPROVED** [review]
Looks great!
You will need to merge the main branch into this branch.
I think the steps will be:
1) checkout to the main branch
2) git pull to update that branch to the newest version
3) now checkout the branch for this pull request
4) git merge main
You will now have to deal with any conflicts in that merge.  I am not an expert, but happy to help.

**Review by edotau — APPROVED** [review]

**Review by craiglowe — APPROVED** [review]
Great to see this!

**Review by cfauci — APPROVED** [review]

**Pull request merged**
